mud-contracts/world/src/modules/smart-gate/systems/SmartGateSystem.sol

```diff
@@ -134,6 +154,10 @@ contract SmartGateSystem is EveSystem, AccessModified {
       revert SmartGate_NotWithtinRange(sourceGateId, destinationGateId);
     }
 
+    //Delete the existing records for the source and destination gate before creating a new link to avoid replacing the record
+    _deleteExistingLink(sourceGateId);
+    _deleteExistingLink(destinationGateId);
```

Comment:
Do we have to call this twice if a single call deletes both source and destination link?

Reply:
In that case `_deleteExistingLink` will have duplicate code. 

eg: 
1. SourceGateId can have 2 links. So 2 record deletion 
2. destinationGateId can have 2 links. So 2 record deletion 

So created a common function `_deleteExistingLink` and calling this twice instead of writing the same code.